src/main/java/org/mycore/externalstore/proxy/MCRExternalStoreProxy.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * This file is part of ***  M y C o R e  ***
+ * See http://www.mycore.de/ for details.
+ *
+ * MyCoRe is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MyCoRe is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MyCoRe.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.mycore.externalstore.proxy;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URL;
+import java.util.Optional;
+
+import org.apache.http.client.utils.URIUtils;
+import org.mitre.dsmiley.httpproxy.ProxyServlet;
+import org.mycore.datamodel.metadata.MCRObjectID;
+import org.mycore.externalstore.MCRExternalStoreService;
+
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+/**
+ * Proxies path to download url.
+ */
+public class MCRExternalStoreProxy extends ProxyServlet {
+
+    private static final long serialVersionUID = 1L;
+
+    @Override
+    protected void initTarget() throws ServletException {
+        // Nothing to do
+    }
+
+    @Override
+    protected void service(HttpServletRequest servletRequest, HttpServletResponse servletResponse)
+        throws ServletException, IOException {
```

Comment:
It would be good if you make a method to extract the Objects from PathInfo and also handle wrong paths with the right http code.